pkg/core/blockchain.go

```diff
@@ -477,6 +477,25 @@ func (bc *Blockchain) init() error {
 	return bc.updateExtensibleWhitelist(bHeight)
 }
 
+func (bc *Blockchain) removeOldStorageItems() {
+	stage, err := bc.dao.Store.Get(storage.SYSStateJumpStage.Bytes())
+	if err != nil || len(stage) == 0 || stage[0] < byte(newStorageItemsAdded) {
+		return
+	}
+
+	b := bc.dao.Store.Batch()
+	prefix := statesync.TemporaryPrefix(bc.dao.Version.StoragePrefix)
+	bc.dao.Store.Seek([]byte{byte(prefix)}, func(k, _ []byte) {
+		// #1468, but don't need to copy here, because it is done by Store.
+		b.Delete(k)
+	})
+
+	err = bc.dao.Store.PutBatch(b)
+	if err != nil {
+		bc.log.Error("failed to remove old storage items", zap.Error(err))
```

Comment:
log.Fatal?

Reply:
It'd be strange to see this error, but the node can continue I think, so why fatal?

Reply:
Because if we're going to support [this](https://github.com/nspcc-dev/neo-go/issues/2156#issuecomment-912583660) then we need *absolutely clean storage* for the second (non-current) prefix.

Reply:
I propose to use `SYSCleanStorage` storing info about the need to remove old items. It can also be used to remove items synchronously for #2156 .